services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainMetrics.java

```diff
@@ -181,17 +181,16 @@ public BeaconChainMetrics(
             "previous_epoch_total_weight",
             "Total effective balance of all active validators in the previous epoch");
 
-    final String versionLabel =
-        VersionProvider.IMPLEMENTATION_VERSION
-            .replaceAll("[\\.\\+]", "_")
-            .replaceAll("[^0-9A-Za-z_]", "");
+    final String version =
+        VersionProvider.IMPLEMENTATION_VERSION.replaceAll("^v", "").replaceAll("\\+.*$", "");
```

Comment:
I think stripping the bit after `+` here is wrong and quite misleading.  For an actual release build the version reported will just be `21.1.1` - only development builds contain the `+` and git hash.  These should not be reported as `21.1.1` as they include additional changes on top of that release and don't have a nice simple version.

Reply:
The alternative would be that when `IMPLEMENTATION_VERSION` contains `+` the `version` label would be set to `develop`.  That would then match the docker label and release filenames (e.g. latest `teku-develop.tar.gz` would give ` 21.2.0+44-g962bff1` if you ran `teku --version`.  The 21.2.0 release would be in `teku-21.2.0.tar.gz` and report `21.2.0`

Reply:
If I understand correctly then `IMPLEMENTATION _VERSION` would look something like `v21.2.0+44-g962bff1` for a development release and `v21.2.0` for a production release, is that correct?

If so, would you want the info to be something like:

```
beacon_teku_version{version="21.2.0",release="v21.2.0"} 1
```

for a released version and:

```
beacon_teku_version{version="develop",release="v21.2.0+45-ge13fd6e"} 1
```

for a development build?

Reply:
Yes that's the alternative. Personally I think I'd just have a single `version` field which would strip the `v` and leave everything else.  So for a release you'd have:
```
beacon_teku_version{version="21.2.0"} 1
```

and for a development build you'd have:
```
beacon_teku_version{version="21.2.0+45-ge13fd63"} 1
```

That's an accurate reflection of what build is actually running in as succinct a way as can be reported.  I'm happy to go either way though, just don't want to have `version` suggest the user is running a released version when they're actually on a completely different development build.  Personally I don't see `develop` as being useful to report as a version but don't mind having it if you see some use for it.

Reply:
I'm fine with your guidance.  I'll open up a new PR with these changes.